### PR TITLE
SendCommands return early if commands len == 0

### PR DESCRIPTION
**Pull request by steiler**
fixing #133

### CODE REVIEW

**Comment by carlmontanari** [comment]
👋 

Thanks @steiler ! I wonder if this should actually return an error instead? Or failing that, it's probably worth making sure the default settings in the MultiResponse all make sense (as in make sense to whatever degree they can when there are no responses stored in it).

Also I didn't check yet but I assume this will be a problem in both Commands/Configs -- probably we should sort both out ya? I can do that but probably wont get to it till the weekend.

Thanks again for raising this and the pr!! 🔥

**Comment by steiler** [comment]
Yeah wasn't sure... An error seems odd because everything is fine it is just an empty configuration file being applied. Basically that should easily go through in my view. 
Not sure what it means for everything else I leave that for you to sort out.
Thanks

**Comment by hellt** [comment]
IMHO returning error is what I'd do. Empty config payload is a no-op which makes no sense from the use case perspective, nor it bears any meaning for testing.
Catching this early with returning a clear error prevent unnecessary rpcs from being made.

**Comment by carlmontanari** [comment]
> IMHO returning error is what I'd do

I dig it.. will add a ErrNoOp or something like that error so it's very clear what happened. Will sort it this weekend if I dont get it done earlier. Thanks as always @hellt !!

**Comment by carlmontanari** [comment]
hey ya'll -- pushed the no-op error thing. lemme know what ya think? if everyones happy ill get that merged!

**Comment by steiler** [comment]
Hi Carl,
I've already worked around it, now checking the length up front.
So whatever you do, except for causing a panic is fine ;-)

File: driver/generic/sendcommands.go [hunk 1]
```diff
@@ -12,13 +14,21 @@ func (d *Driver) SendCommands(
...
 	if err != nil {
 		return nil, err
 	}
 
 	m := response.NewMultiResponse(d.Transport.GetHost())
 
+	if len(commands) == 0 {
+		return m, err
+	}
+
```
Comment (hellt) [inline driver/generic/sendcommands.go:31]: @carlmontanari 
unless I am missing something I don't see why this check is here, while you have it on Ln17 already?

Comment (carlmontanari) [reply driver/generic/sendcommands.go:31]: ahaha no, as per usual not you missing things -- just me :D ty ty, fixing and will merge!

**Pull request merged**